tests/integration/workspace_access/test_groups.py

```diff
@@ -99,6 +100,8 @@ def test_delete_ws_groups_should_delete_renamed_and_reflected_groups_only(
     group_manager.reflect_account_groups_on_workspace()
     group_manager.delete_original_workspace_groups()
 
+    time.sleep(5)  # It takes a moment to delete the group through the API
```

Comment:
never sleep explicitly, it's a recipe for slowdown. use `@retried(on=[NotFound], timeout=timedelta(seconds=10))` decorator instead

Reply:
Good, I like to use the retried. Five seconds felt reasonable to use sleep instead of an involved polling. But the retried hides that well